server/service/appconfig.go

```diff
@@ -333,6 +333,9 @@ func validateSSOSettings(p fleet.AppConfig, existing *fleet.AppConfig, invalid *
 				invalid.Append("idp_name", "required")
 			}
 		}
+		if license.Tier != "premium" && p.SSOSettings.EnableJITProvisioning {
```

Comment:
I think this should use `license.IsPremium()`, which is less error-prone but also checks for backward compatibility with an old tier name.

Reply:
ohh thanks for this! didn't know and copied a conditional from above, I updated both